server/src/test/java/org/elasticsearch/snapshots/MinThreadsSnapshotRestoreIT.java

```diff
@@ -0,0 +1,208 @@
+import io.crate.testing.UseJdbc;
+import org.elasticsearch.action.ActionFuture;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.repositories.RepositoriesService;
+import org.elasticsearch.snapshots.mockstore.MockRepository;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.containsString;
+
+/**
+ * Tests for snapshot/restore that require at least 2 threads available
+ * in the thread pool (for example, tests that use the mock repository that
+ * block on master).
+ */
+@UseJdbc(0)
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class MinThreadsSnapshotRestoreIT extends AbstractSnapshotIntegTestCase {
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder().put(super.nodeSettings(nodeOrdinal))
+            .put("thread_pool.snapshot.core", 2)
+            .put("thread_pool.snapshot.max", 2)
+            .build();
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        var plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(MockRepository.Plugin.class);
+        return plugins;
+    }
+
+    public void testConcurrentSnapshotDeletionsNotAllowed() throws Exception {
+        logger.info("--> creating repository");
+
+        assertAcked(client().admin().cluster().preparePutRepository("repo").setType("mock").setSettings(
+            Settings.builder()
+                .put("location", randomRepoPath())
+                .put("random", randomAlphaOfLength(10))
+                .put("wait_after_unblock", 200)).get());
+
+        logger.info("--> snapshot twice");
+        execute("create table doc.test1 (x integer) clustered into 1 shards with (number_of_replicas=0)");
+        for (int i = 0; i < 10; i++) {
+            execute("insert into doc.test1 values(?)", new Object[]{i});
+        }
+        refresh();
+        execute("create snapshot repo.snapshot1 all WITH (wait_for_completion=true)");
+
+        execute("create table doc.test2(x integer) clustered into 1 shards with (number_of_replicas=0)");
+
+        for (int i = 0; i < 10; i++) {
+            execute("insert into doc.test2 values(?)", new Object[]{i});
+        }
+        refresh();
+        execute("create snapshot repo.snapshot2 all WITH (wait_for_completion=true)");
+
+        String blockedNode = internalCluster().getMasterName();
+        ((MockRepository)internalCluster().getInstance(RepositoriesService.class, blockedNode).repository("repo")).blockOnDataFiles(true);
+        logger.info("--> start deletion of first snapshot");
+        ActionFuture<AcknowledgedResponse> future =
```

Comment:
This log is not correct, the test deletes the second snapshot. However this is exactly like in the original test.